test/jdk/java/foreign/TestMatrix.java

```diff
@@ -1,8 +1,8 @@
 /*
- * @test
+ * @test id=UpcallHighArity-FFTT
  * @requires ((os.arch == "amd64" | os.arch == "x86_64") & sun.arch.data.model == "64") | os.arch == "aarch64"
  * @modules jdk.incubator.foreign/jdk.internal.foreign
- * @build NativeTestHelper CallGeneratorHelper TestUpcallHighArity TestUpcall TestDowncall
+ * @build NativeTestHelper CallGeneratorHelper TestUpcallHighArity
  *
  * @run testng/othervm/native
```

Comment:
can you please throw in a `/manual` in here as well - so that this test will be disabled when running tests with jtreg `-a` option (which should be the norm in headless mode).

Reply:
I can, but it seems to me that `/manual` is for marking GUI tests that require user interaction. Let me see if there are keywords we can use. Something like "developer", so that you can pass `jtreg -k:developer` or `JTREG_KEYWORDS=developer make test ...` when you really want to run them. I'll take a closer look tomorrow.

Reply:
you are right about /manual being for GUI - but @AlanBateman pointed me at few tests with libzip using it for similar reasons (e.g. long running tests).

Reply:
I don't think /manual is strictly UI but its usage is discouraged as manual tests are rarely run.  The tests for the SmartCard I/O API come to mind as they need special setup. There is one or two ZIP tests that take a long time and have historically been manual tests as a result. There are also a few tests dotted around that do not have the `@Test` tag, one needs to be working in a specific area to know about these. Introducing a new test group or a keyword for these tests could be an option.

In passing, I should mention the jdk_sctp test group. These tests may require kernel or other configuration and are deliberately not in any tier. That may be a comment for the other PR that is proposing a jdk tier4.

Reply:
See new commit.